gaia-component.js

```diff
@@ -147,6 +83,26 @@ var base = {
     },
 
     /**
+     * A convenient method for triggering l10n for component's shadow DOM.
+     */
+    setupShadowL10n: function() {
+      if (!document.l10n) { return; }
+      this.onDOMRetranslated = this.localizeShadow.bind(null, this.shadowRoot);
+      document.l10n.ready.then(() => {
+        document.addEventListener('DOMRetranslated', this.onDOMRetranslated);
+        this.localizeShadow(this.shadowRoot);
+      });
+    },
+
+    /**
+     * Localizes the shadowRoot subtree.
+     */
+    localizeShadow: function(shadowRoot) {
+      if (!document.l10n) { return; }
+      document.l10n.translateFragment(shadowRoot);
+    },
```

Comment:
Yeah good point, I updated the setupShadowL10n slightly where instead of just returning when no l10n is present, we still call localizeShadow that might've been customized by the component author.